app/api/route.js

```diff
@@ -0,0 +1,298 @@
+      const where = searchName
+        ? {
+            name: {
+              [Op.iLike]: `%${searchName}%`,
+            },
+          }
+        : {};
+
+      const { rows, count } = await Author.findAndCountAll({
+        where,
+        limit: size,
+        offset: offset,
+      });
+
+      const validatedAuthors = await Promise.all(
+        rows.map(async (author) => ({
+          ...author.toJSON(),
+          profilePhotoUrl: await validateImageUrl(
+            author.profilePhotoUrl,
+            "author"
+          ),
+        }))
+      );
+
+      return {
+        authors: validatedAuthors,
+        totalCount: count,
+        hasMore: offset + rows.length < count,
+      };
+    },
+    book: async (parent, { id }, context, info) => {
+      const book = await Book.findByPk(id);
+      if (!book) {
+        throw new Error("Book not found");
+      }
+      return book;
+    },
+    author: async (parent, args, context, info) => {
+      return Author.findByPk(args.id);
+    },
+    allAuthors: async () => {
+      return await Author.findAll();
+    },
+  },
+  Book: {
+    author: async (parent, args, context, info) => {
+      return batchAuthors.load(parent.author_id);
+    },
+  },
+  Author: {
+    books: async (parent, args, context, info) => {
+      return batchBooks.load(parent.id);
+    },
+  },
+  Mutation: {
+    addBook: async (parent, args, context, info) => {
+      return await Book.create(args, { returning: true });
+    },
+    addAuthor: async (parent, args, context, info) => {
+      const result = await Author.create(args, { returning: true });
+      return result;
+    },
+    updateBook: async (parent, args, context, info) => {
+      if (args.publishedAt) {
+        args.publishedAt = moment(args.publishedAt);
+      }
+      return (
+        await Book.update(args, {
+          where: { id: args.id },
+          returning: true,
+        })
+      )[1][0];
+    },
+    updateAuthor: async (parent, args, context, info) => {
+      try {
+        if (args.date_of_birth) {
+          args.date_of_birth = moment(args.date_of_birth);
+        }
+        let result = await Author.update(args, {
+          where: { id: args.id },
+          returning: true,
+        });
+        return result[1][0];
+      } catch (error) {
+        console.log(error);
+        return null;
+      }
+    },
+    deleteBook: async (parent, args, context, info) => {
+      return await Book.destroy({ where: { id: args.id }, returning: true });
+    },
+    deleteAuthor: async (parent, args, context, info) => {
+      return await Author.destroy({ where: { id: args.id } });
+    },
+    addReview: async (_, { bookId, rating, comment }) => {
+      try {
+        const book = await Book.findByPk(bookId);
+        if (!book) {
+          throw new Error("Book not found");
+        }
+        const reviews = book.toJSON().reviews || [];
+        reviews.push({ rating, comment });
+        await book.update({ reviews });
```

Comment:
we should add a seperate field for reviews, and resolve it seperately, instead of doing this in the mutation